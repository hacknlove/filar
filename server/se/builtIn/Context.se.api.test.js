const { DOMParser } = require("linkedom");
const { processAllElements } = require("../../tree/processAllElements");
const { ServerElementsMap } = require("../common");
const parser = new DOMParser();

const Context = require("./Context.se");

describe("Context from api", () => {
  beforeAll(() => {
    global.fetch = jest.fn();
  });
  afterAll(() => {
    delete global.fetch;
  });

  it("changes parent context from a get API call", async () => {
    const document = parser.parseFromString(`
            <div Title="'My title'">
                <Context src="//example.com/context"></Context>
                <h1>{{Title}}</h1>
            </div>
        `);

    fetch.mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve({ Title: "My Updated title" }),
      })
    );

    ServerElementsMap.set("Context", Context);

    await processAllElements(document, {
      filePath: "test/index.html",
      __islands: {},
      __ce: {},
    });

    expect(document.toString()).toMatchSnapshot();
  });
});
