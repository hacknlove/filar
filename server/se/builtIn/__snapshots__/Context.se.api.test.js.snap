// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context from api changes parent context from a get API call 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
            <div>
                
                <h1>My Updated title</h1>
            </div>
        "
`;
